src/output-json.c

```diff
@@ -694,6 +694,29 @@ void JsonTcpFlags(uint8_t flags, json_t *js)
         json_object_set_new(js, "cwr", json_true());
 }
 
+/** \brief jsonify tcp flags field
+ *  Only add 'true' fields in an attempt to keep things reasonably compact.
+ */
+void EveTcpFlags(uint8_t flags, JsonBuilder *js)
+{
+    if (flags & TH_SYN)
+        jb_set_bool(js, "syn", true);
+    if (flags & TH_FIN)
+        jb_set_bool(js, "fin", true);
+    if (flags & TH_RST)
+        jb_set_bool(js, "rst", true);
+    if (flags & TH_PUSH)
+        jb_set_bool(js, "psh", true);
+    if (flags & TH_ACK)
+        jb_set_bool(js, "ack", true);
+    if (flags & TH_URG)
+        jb_set_bool(js, "urg", true);
```

Comment:
we could possibly micro-optimize calls like this as it produces a constant output `urg: true`

Reply:
In C at least, maybe a macro that just creates a constant string?

Untested:
```c
#define JB_SET_PREFORMATTED(jb, key, value) jb_add((jb), (key) ":" (value))
```

Reply:
Any thoughts on this?

Reply:
This is what I've come up with that seems to work..  The Rust method is:
```
jb_set_formatted(JsonBuilder *js, const char *formatted);
```

To use with a bool like above from C:
```
jb_set_formatted(jb, "\"ack\":true");
```

I've also added a macro:
```
#define JB_SET_FORMATTED(jb, key, val) jb_set_formatted((jb), "\"" key "\":" val)
```
so the above would be:
```
JB_SET_FORMATTED(jb, "ack", "true");
```
however, to deal with https://github.com/OISF/suricata/pull/5030#discussion_r436290712, we'd have to do something like:
```
JB_SET_FORMATTED(jb, "command", "\"RETR\"");
```
but I set we could have JB_SET_STRING, JB_SET_TRUE, JB_SET_FALSE to handle some of these details.

Reply:
As a PR: https://github.com/OISF/suricata/pull/5042